weapons-on-back/cl_weapons-on-back.lua

```diff
@@ -40,27 +41,34 @@ local SETTINGS = {
     }
 }
 
+--- Other Variables ---
 local attached_weapons = {}
+local me = GetPlayerPed(-1)
+local bone = GetPedBoneIndex(GetPlayerPed(-1), boneNumber)
+local toggle = (1,19) --- Change toggle key here. Currently: left alt.
 
-Citizen.CreateThread(function()
-  while true do
-      local me = GetPlayerPed(-1)
-      ---------------------------------------
-      -- attach if player has large weapon --
-      ---------------------------------------
-      for wep_name, wep_hash in pairs(SETTINGS.compatable_weapon_hashes) do
-          if HasPedGotWeapon(me, wep_hash, false) then
-              if not attached_weapons[wep_name] then
+
+if (IsControlJustReleased(toggle)) then --- Toggle here
+      ------------------------------------------------
+      -- attaches weapon if player has large weapon --
+      ------------------------------------------------
+    for wep_name, wep_hash in pairs(SETTINGS.compatable_weapon_hashes) do
+        if HasPedGotWeapon(me, wep_hash, false) then
+            if not attached_weapons[wep_name] then
                   AttachWeapon(wep_name, wep_hash, SETTINGS.back_bone, SETTINGS.x, SETTINGS.y, SETTINGS.z, SETTINGS.x_rotation, SETTINGS.y_rotation, SETTINGS.z_rotation, isMeleeWeapon(wep_name))
-              end
-          end
-      end
-      --------------------------------------------
-      -- remove from back if equipped / dropped --
-      --------------------------------------------
-      for name, attached_object in pairs(attached_weapons) do
-          -- equipped? delete it from back:
-          if GetSelectedPedWeapon(me) ==  attached_object.hash or not HasPedGotWeapon(me, attached_object.hash, false) then -- equipped or not in weapon wheel
+            end
+        end
+    end
+end
+
+Wait(100) -- this just makes sure that toggle doesnt overlap

```

Comment:
This would go in the loop as well and probably leaving it as 0 would be best so that it is more responsive to user inputs.

Reply:
Yes, belongs in the loop,  my mistake, made it to 1, since i still need a small time value.